applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/snats.js

```diff
@@ -235,6 +235,10 @@ return view.extend({
 		o.depends({ proto: 'tcp', '!contains': true });
 		o.depends({ proto: 'udp', '!contains': true });
 
+		o = s.taboption('advanced', form.Value, 'ipset', _('Use ipset'), _('Unsupported in fw4'));
```

Comment:
Please wrap this option declaration in `if (!L.hasSystemFeature('firewall4')) { ... }` and remove the `Unsupported in fw4` remark.

Reply:
We should loop uci sections of type `ipset` and present their names here, like

```javascript
o = s.taboption('advanced', form.Value, 'ipset', _('Use ipset'));
uci.sections('firewall', 'ipset', function(s) {
    if (typeof(s.name) == 'string')
            this.value(s.name, s.comment ? '%s (%s)'.format(s.name, s.comment) : s.name);
});
```